src/ripple/app/rdb/backend/RelationalDBInterfaceSqlite.cpp

```diff
@@ -173,6 +175,7 @@ class RelationalDBInterfaceSqliteImp : public RelationalDBInterfaceSqlite
 
 private:
     Application& app_;
+    bool useTxTables_;
```

Comment:
The issue is that the code assumes that at least one of the transaction db or the shard db will be present, and recent changes means that both databases may be absent.

Instead of a `useTxTables_` flag, what about structuring the code so the calls look like:
```
if (existsTransaction()){
   return withTxDb();
}
else if (existsShard()){
   return withShardsDb();
}
return default; // std::nullopt or somesuch
```

Reply:
I don't know what could be implementation of existsShard(). Each request to shards as a rule will touch several shards databases. Also need to take into account what means option use_tx_tables = 0. Is it prohibit use of node transaction databases only, or overall transactions databases which includes shard transaction databases too. In the last case my implementation looks more correct.

Reply:
I thought `existsShard()` was just `app_.getShardStore() != nullptr`. But you're right, we need to take into account what `use_tx_tables = 0` means, and I agree your implementation looks more correct. Let's keep yours.

Reply:
This is what use_tx_tables = 0 means (from cfg/rippled-example.cfg):
#      use_tx_tables        Valid values: 1, 0
#                           The default is 1 (true). Determines whether to use
#                           the SQLite transaction database. If set to 0,
#                           rippled will not write to the transaction database,
#                           and will reject tx, account_tx and tx_history RPCs.
#                           In Reporting Mode, this setting is ignored.

It doesn't stop the use of the shard transaction databases though.

Reply:
I'll defer to @mtrippled and @cjcobb23 as to what the correct behaviour should be for `use_tx_tables = 0` for this class, since they are more familiar with how that switch is mean to be used.

Reply:
The correct behavior should be to not write to the transaction databases at all if use_tx_tables is 0. This includes the node transaction database and the shard transaction database.

Reply:
In second patch of changes I based on the last point of view. But for alternative point it need to change one line in the code.